pkg/cache/tas_flavor_snapshot.go

```diff
@@ -738,6 +823,35 @@ func useLeastFreeCapacityAlgorithm(unconstrained bool) bool {
 		(unconstrained && features.Enabled(features.TASProfileMixed))
 }
 
+func (s *TASFlavorSnapshot) updateSliceCountsToMinimum(domains []*domain, count int32, sliceSize int32, unconstrained bool) []*domain {
+	result := make([]*domain, 0)
+	remainingCount := count
+	for i, domain := range domains {
+		if useBestFitAlgorithm(unconstrained) && domain.sliceState >= remainingCount/sliceSize {
+			// optimize the last domain
+			mostAllocatedIdx := 0
+			mostAllocatedIdx = findBestFitDomainIdxForSlices(domains[i:], remainingCount/sliceSize)
+			domain = domains[i+mostAllocatedIdx]
+		}
```

Comment:
Can this be pulled out of the for-loop? I think that we will always return after this call, since we check that condition before entering this clause.

Reply:
I would leave it as it is for this PR, because that's consistent with the other function that was here before (`updateCountsToMinimum`), so I would like to improve the old one as well. I'll add it as a thing to consider in the follow up PR. WDYT?